tmol/score/ljlk/LJLKEnergy.py

```diff
@@ -0,0 +1,349 @@
+        heavy_atoms_in_tile = torch.full(
+            (packed_block_types.n_types, max_n_tiles * self.tile_size),
+            -1,
+            dtype=torch.int32,
+            device=self.device,
+        )
+        n_heavy_ats_in_tile = torch.full(
+            (packed_block_types.n_types, max_n_tiles),
+            0,
+            dtype=torch.int32,
+            device=self.device,
+        )
+
+        def _t(arr):
+            return torch.tensor(arr, dtype=torch.int32, device=self.device)
+
+        for i, rt in enumerate(packed_block_types.active_block_types):
+            i_n_tiles = rt.ljlk_n_heavy_atoms_in_tile.shape[0]
+            i_n_tile_ats = i_n_tiles * self.tile_size
+            heavy_atoms_in_tile[i, :i_n_tile_ats] = _t(rt.ljlk_heavy_atoms_in_tile)
+            n_heavy_ats_in_tile[i, :i_n_tiles] = _t(rt.ljlk_n_heavy_atoms_in_tile)
+
+        setattr(packed_block_types, "ljlk_heavy_atoms_in_tile", heavy_atoms_in_tile)
+        setattr(packed_block_types, "ljlk_n_heavy_atoms_in_tile", n_heavy_ats_in_tile)
+
+    def setup_poses(self, poses: PoseStack):
+        super(LJLKEnergy, self).setup_poses(poses)
+
+    def inter_module(
+        self,
+        packed_block_types: PackedBlockTypes,
+        systems: PoseStack,
+        context_system_ids: Tensor[int][:, :],
+        system_bounding_spheres: Tensor[float][:, :, 4],
+        weights,  # map string->Real
+    ):
+        system_neighbor_list = self.create_block_neighbor_lists(
+            systems, system_bounding_spheres
+        )
+        lj_lk_weights = torch.zeros((2,), dtype=torch.float32, device=self.device)
+        lj_lk_weights[0] = weights["lj"] if "lj" in weights else 0
+        lj_lk_weights[1] = weights["lk"] if "lk" in weights else 0
+
+        pbt = packed_block_types
+        return LJLKInterSystemModule(
+            context_system_ids=context_system_ids,
+            system_min_block_bondsep=systems.min_block_bondsep,
+            system_inter_block_bondsep=systems.inter_block_bondsep,
+            system_neighbor_list=system_neighbor_list,
+            bt_n_atoms=pbt.n_atoms,
+            bt_n_heavy_atoms=pbt.n_heavy_atoms,
+            bt_n_heavy_atoms_in_tile=pbt.ljlk_n_heavy_atoms_in_tile,
+            bt_heavy_atoms_in_tile=pbt.ljlk_heavy_atoms_in_tile,
+            bt_atom_types=pbt.atom_types,
+            bt_heavy_atom_inds=pbt.heavy_atom_inds,
+            bt_n_interblock_bonds=pbt.n_interblock_bonds,
+            bt_atoms_forming_chemical_bonds=pbt.atoms_for_interblock_bonds,
+            bt_path_distance=pbt.bond_separation,
+            type_params=self.type_params,
+            global_params=self.global_params,
+            lj_lk_weights=lj_lk_weights,
+        )
+
+    def create_block_neighbor_lists(
+        self, systems: PoseStack, system_bounding_spheres: Tensor[float][:, :, 4]
+    ):
+        # we need to make lists of all block pairs within
+        # striking distances of each other that we will use to
```

Comment:
fine for now but in the future should think about moving most of this logic to common functions